src/service/resource.js

```diff
@@ -396,7 +412,12 @@ export class Resource {
    * @return {!../layout-rect.LayoutRectDef}
    */
   getLayoutBox() {
-    return this.layoutBox_;
+    if (!this.isFixed_) {
+      return this.layoutBox_;
+    }
+    const viewport = this.resources_.getViewport();
+    return moveLayoutRect(this.layoutBox_, viewport.getScrollLeft(),
```

Comment:
Hmm. This is a bit troublesome. The definition of the `getLayoutBox` is that it doesn't change based on the scroll state. We are de-facto changing it here. I'm not sure what downstream issues we maybe expecting from this. The truth is, of course, that fixed element's layout box doesn't change but defined within another (fixed) layer. I think a better option for us would be:

1. Modify `overlaps` method to incorporate the knowledge of `isFixed`. We can even rename it as `overlapsViewport`. IMO, this is better b/c this is a more direct API that's only used for the purposes of scheduling.
2. At this time, propagating `getLayoutBox` is rather dangerous. I don't know who uses it and where. I think it's best to replace this call with `getLayoutSize()` or `getLayoutWidth/Height` pair. I believe most of clients only care about size and not absolute position.